src/extension/module.js

```diff
@@ -251,6 +251,40 @@
     prod: 'host',
   };
 
+/**
+ * log RUM if part of the sample.
+ * @param {string} checkpoint identifies the checkpoint in funnel
+ * @param {Object} data additional data for RUM sample
+ */
+
+// eslint-disable-next-line import/prefer-default-export
+function sampleRUM(checkpoint, collect = false, data = {}) {
+  try {
+    window.hlx = window.hlx || {};
+    if (!window.hlx.rum) {
+      const usp = new URLSearchParams(window.location.search);
+      const weight = 5; 
+      // eslint-disable-next-line no-bitwise
+      const hashCode = (s) => s.split('').reduce((a, b) => (((a << 5) - a) + b.charCodeAt(0)) | 0, 0);
+      const id = `${hashCode(window.location.href)}-${new Date().getDay()}-${new Date().getHours()}-${chrome.runtime.id}`;
+      const random = Math.random();
+      const isSelected = (random * weight <= (collect) ? 5 : 1);
```

Comment:
Why the 20% sampling rate on the authoring side? In our case we would like to know about trends in terms of Avg. time to market. The sampling rate makes sense from the end user perspective because of volume of requests etc, but from the authoring side I would want to know the average of all the times someone has gone from 0->Live. Curious to get your thoughts on this. @MarquiseRosier

Reply:
Yes, thanks for this pointer @Amol! I made sure to ensure all publishing/opening is collected!

Reply:
sampling is a [core principle](https://en.wikipedia.org/wiki/Privacy_by_design) of our RUM collection. It ensures privacy and reduces operational cost. We can discuss the sampling *rate*, which I think is appropriate at 20% for authoring, but not the fact that we are sampling – and should we discover that customers are abusing the variable sampling rate mechanism, we will drop these samples on the server-side.

For the example mentioned by @amol-anand, a full sample isn't even necessary, this can be estimated using our existing data (just measure the time of the first request on an owner/repo/ref combination vs. the point where BYO CDN requests make up the majority of traffic.